frontend/src/app/components/api/op-file-upload/op-direct-file-upload.service.ts

```diff
@@ -100,13 +101,16 @@ export class OpenProjectDirectFileUploadService extends OpenProjectFileUploadSer
     };
   }
 
-  public getDirectUploadFormFrom(url:string, file:UploadFile|UploadBlob):Promise<PrepareUploadResult> {
+  public async getDirectUploadFormFrom(url:string, file:UploadFile|UploadBlob):Promise<PrepareUploadResult> {
+    const fileName = file.customName || file.name;
+    const contentType = fileName ? lookup(fileName) || file.type : file.type;
```

Comment:
Prefer to use the browser's provided content type, as I'd image that would be more reliable. We can fall back to mime-type if that is empty.

```suggestion
    let contentType = fileName || (fileName && lookup(fileName)) || '';
```

Reply:
Makes sense. 👍